agent/dockerclient/dockerapi/docker_client_test.go

```diff
@@ -908,104 +908,161 @@ func TestUnavailableVersionError(t *testing.T) {
 }
 
 func TestStatsNormalExit(t *testing.T) {
-	mockDocker, _, client, _, _, _, done := dockerClientSetup(t)
+	_, mockDockerSDK, client, _, _, _, done := dockerClientSetup(t)
 	defer done()
-	time1 := time.Now()
-	time2 := time1.Add(1 * time.Second)
-	mockDocker.EXPECT().Stats(gomock.Any()).Do(func(x interface{}) {
-		opts := x.(docker.StatsOptions)
-		defer close(opts.Stats)
-		assert.Equal(t, "foo", opts.ID, "Expected ID foo, got %s", opts.ID)
-		assert.True(t, opts.Stream, "Expected stream to be true")
-		opts.Stats <- &docker.Stats{
-			Read: time1,
-		}
-		opts.Stats <- &docker.Stats{
-			Read: time2,
-		}
-	})
+	mockDockerSDK.EXPECT().ContainerStats(gomock.Any(), gomock.Any(), true).Return(types.ContainerStats{
+		Body: mockStream{
+			data:  []byte(`{"memory_stats":{"Usage":50},"cpu_stats":{"system_cpu_usage":100}}`),
+			index: 0,
+		},
+	}, nil)
 	ctx, cancel := context.WithCancel(context.TODO())
 	defer cancel()
-	stats, err := client.Stats("foo", ctx)
+	stats, err := client.Stats(ctx, "foo", StatsInactivityTimeout)
 	assert.NoError(t, err)
-	stat := <-stats
-	checkStatRead(t, stat, time1)
-	stat = <-stats
-	checkStatRead(t, stat, time2)
-	stat = <-stats
-	assert.Nil(t, stat, "Expected stat to be nil")
-}
-
-func checkStatRead(t *testing.T, stat *docker.Stats, read time.Time) {
-	assert.Equal(t, read, stat.Read, "Expected %v, but was %v", read, stat.Read)
+	newStat := <-stats
+	waitForStats(t, newStat)
+	assert.Equal(t, uint64(50), newStat.MemoryStats.Usage)
+	assert.Equal(t, uint64(100), newStat.CPUStats.SystemUsage)
 }
 
-func TestStatsClosed(t *testing.T) {
```

Comment:
why is this test deleted?

Reply:
this as unit test doesn't make sense any more with SDK; with go-dockerclient it returned stream for us to manipulate, whereas with SDK it is returning reader wrapped in a response object.